docs/source/reference/prototype.rst

```diff
@@ -21,47 +25,132 @@ item assignment and many more features.
-          fields={
-              floatdata: Tensor(torch.Size([3, 4, 5]), dtype=torch.float32),
-              intdata: Tensor(torch.Size([3, 4, 1]), dtype=torch.int64)},
-          batch_size=torch.Size([3, 4]),
-          device=None,
-          is_shared=False),
-      batch_size=[3, 4],
-      device=None,
-      is_shared=False)
+    floatdata=Tensor(shape=torch.Size([3, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+    intdata=Tensor(shape=torch.Size([3, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+    non_tensordata='test',
+    nested=MyData(
+        floatdata=Tensor(shape=torch.Size([3, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+        intdata=Tensor(shape=torch.Size([3, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+        non_tensordata='nested_test',
+        nested=None,
+        batch_size=torch.Size([3, 4]),
+        device=None,
+        is_shared=False),
+    batch_size=torch.Size([3, 4]),
+    device=None,
+    is_shared=False)
+
+
+:obj:`@tensorclass` supports indexing. Internally the tensor objects gets indexed,
+however the non-tensor data remains same
+
+.. code-block::
+
+
+ >>>print("indexed:", data[:2])
+ indexed: MyData(
+    floatdata=Tensor(shape=torch.Size([2, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+    intdata=Tensor(shape=torch.Size([2, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+    non_tensordata='test',
+    nested=MyData(
+        floatdata=Tensor(shape=torch.Size([2, 4, 5]), device=cpu, dtype=torch.float32, is_shared=False),
+        intdata=Tensor(shape=torch.Size([2, 4, 1]), device=cpu, dtype=torch.int64, is_shared=False),
+        non_tensordata='nested_test',
+        nested=None,
+        batch_size=torch.Size([2, 4]),
+        device=None,
+        is_shared=False),
+    batch_size=torch.Size([2, 4]),
+    device=None,
+    is_shared=False)
+
+:obj:`@tensorclass` also supports and get and set attributes. One can also access
+nested tensor class data
+
+.. code-block::
+
+ >>> data.non_tensordata = "test_changed"
+ >>> print("data.non_tensordata: ", repr(data.non_tensordata))
+ data.non_tensordata: 'test_changed'
+
+ >>> data.floatdata = torch.ones(3, 4, 5)
+ >>>print("data.floatdata:", data.floatdata)
+ data.floatdata: tensor([[[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]],
+
+        [[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]],
+
+        [[1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.],
+         [1., 1., 1., 1., 1.]]])
 
+ >>> # Changing nested tensor data
+ >>> data.nested.non_tensordata = "nested_test_changed"
+ >>>print("data.nested.non_tensordata:", repr(data.nested.non_tensordata))
+ data.nested.non_tensordata: 'nested_test_changed'
+
+:obj:`@tensorclass` supports torch functions and the behavior is extened to
```

Comment:
it does not support all torch functions
we should be clear that it's a subset of torch functions, mostly shape operations, and no algebraic operations.